osmapi/OsmApi.py

```diff
@@ -1963,33 +1973,33 @@ def _http(self, cmd, path, auth, send):  # noqa
                 self._conn = self._get_http_connection()
 
     def _get_http_connection(self):
-        https_str = 'https://'
-        http_str = 'http://'
-        if self._api.lower().startswith(https_str):
-            return httplib.HTTPSConnection(self._api[len(https_str):], 443)
-        elif self._api.lower().startswith(http_str):
-            return httplib.HTTPConnection(self._api[len(http_str):], 80)
-        else:
-            return httplib.HTTPConnection(self._api, 80)
+        """
+        Creates a requests session for connection pooling.
+        """
+        session = requests.Session()
+        session.headers.update({
+            'user-agent': self._created_by
+        })
+        return session
 
     def _sleep(self):
         time.sleep(5)
 
     def _get(self, path):
-        return self._http('GET', path, False, None)
+        return self._http(self._conn.get, path, False, None)
```

Comment:
I haven't seen that method used before, but it is likely an easy change. I should be able to do this in the next ~24 hours.